libsoftwarecontainer/src/gateway/network/networkgateway.cpp

```diff
@@ -85,7 +82,9 @@ bool NetworkGateway::activateGateway()
         return false;
     }
 
-    if (isError(generateIP())) {
+
+    if (false == m_functions.generateIP(m_netmask, m_gateway, m_containerID, m_ip)) {
```

Comment:
If this returns true/false you should probably use `!m_function.generateIP(...)` instead.

However it makes more sense to have such a function return an IP and throw exception in case of error I think.